### PR TITLE
MQTT: handle socket disconnection during reconnection attempts

### PR DESCRIPTION
**Pull request by chriadam**

### CODE REVIEW

File: src/qt/ve_qitems_mqtt.cpp [hunk 1]
```diff
@@ -285,6 +285,7 @@ void VeQItemMqttProducer::transitionState()
 	if (haveTransition) {
 		// transition to the appropriate state, then perform side effects
 		// associated with that state.
+		// qDebug() << "Transitioning from" << mConnectionState << " to " << transition.to;
```
Comment (DanielMcInnes) [inline src/qt/ve_qitems_mqtt.cpp:288]: is this intentional? (it's ok if it is)

Comment (chriadam) [reply src/qt/ve_qitems_mqtt.cpp:288]: Yes, I find myself adding this line of code every time I need to debug a problem.  So, figured I'd just leave it here, but commented out, so in future I can just easily uncomment it while debugging.

File: src/qt/ve_qitems_mqtt.cpp [hunk 2]
```diff
@@ -950,7 +970,8 @@ bool WebSocketDevice::open(QIODeviceBase::OpenMode mode)
 	QWebSocketHandshakeOptions options;
 	options.setSubprotocols(QStringList { QString::fromUtf8(mProtocol) });
 	mWebSocket.open(mUrl, options);
-	return QIODevice::open(mode);
+	mConnecting = QIODevice::open(mode);
```
Comment (DanielMcInnes) [inline src/qt/ve_qitems_mqtt.cpp:973]: it would be good to log something if QIODevice::open() fails and returns false

Comment (chriadam) [reply src/qt/ve_qitems_mqtt.cpp:973]: good point, will do

**Review by DanielMcInnes — APPROVED** [review]

File: src/qt/ve_qitems_mqtt.cpp (hunk 2 above)
Comment (chriadam) [reply src/qt/ve_qitems_mqtt.cpp:989]: done, handled by the caller

**Pull request merged**